### ISSUE TITLE
Migration from 1.3.6 to 1.4.0 might miss apps and eventually kill tasks

### ISSUE THREAD

**Issue by meichstedt** [mesosphere/marathon#4968]
Background: HA setup with 3 masters, all three are subsequentially migrated to 1.4.0

It seems like the guideline of [Upgrading an HA installation](https://mesosphere.github.io/marathon/docs/upgrade/index.html) is not followed correctly and we can see that after migrating the zk state, an old Marathon instance is elected as leader. I don't think that's the issue here but we'll need to close that one out too.

Although the migration succeeds often, there seems to be a race somewhere. The logs of a migration show the following timeline:

```
06:15:58 10.10.0.31   about to upgrade to 1.4.0
06:19:19 10.10.0.31   is starting up Marathon-1.4.0
06:19:21 10.10.0.68   about to upgrade to 1.4.0
06:19:40 10.10.0.31   starts zk-migration to 1.4.0
06:19:41 10.10.0.31   completes zk-migration for version: Version(1, 4, 0, LEGACY)
                      Migrated (tasks,2)(deployment plans,0)(task failures,0)(root + app versions,8)(framework-id,1)(event-subscribers,0) to new format
06:19:41 10.10.0.31   ERROR Mesos Master version 1.0.1 does not meet minimum required version 1.1.0
06:19:41 10.10.0.31   ERROR Committing suicide! 
06:19:46 10.10.0.107  is starting up Marathon-1.3.6
06:19:48 10.10.0.107  Elected as leader
06:22:37 10.10.0.107  about to upgrade to 1.4.0
06:22:39 10.10.0.68   is starting up Marathon-1.4.0
06:22:48 10.10.0.68   Elected as leader
06:22:49 10.10.0.68   INFO  No migration necessary, already at the current version
06:23:04 10.10.0.68   initiates a task reconciliation
06:23:04 10.10.0.68   WARN  RunSpec /upgrade-healthcheck-68d3ee04be9d4b4f928c4f4aba7df65a exists in InstanceTracker, but not store. The run spec was likely terminated. Will now expunge.
06:23:04 10.10.0.68   INFO  Killing 1 tasks for reason: Orphaned (ids: instance [upgrade-healthcheck-68d3ee04be9d4b4f928c4f4aba7df65a.marathon-403c3bca-dd45-11e6-8fbb-02425b5a8329] ...)
06:23:04 10.10.0.68   WARN  RunSpec /upgrade-dns-68d3ee04be9d4b4f928c4f4aba7df65a exists in InstanceTracker, but not store. The run spec was likely terminated. Will now expunge.
06:23:04 10.10.0.68   INFO  Killing 1 tasks for reason: Orphaned (ids: instance [upgrade-dns-68d3ee04be9d4b4f928c4f4aba7df65a.marathon-59ab76cb-dd45-11e6-8fbb-02425b5a8329] ...)
```

**Comment by meichstedt** [comment]
See logs in [DCOS-13204](https://mesosphere.atlassian.net/browse/DCOS-13204)

**Comment by aquamatthias** [comment]
We need to check if we can read the zk format. 
During an upgrade it can happen, that a Migration to 1.4 is already applied and a 1.3 Leader takes over. This leader needs to abdicate in that case.

We have this logic in 1.4 but not in 1.3.

**Comment by jasongilanfarr** [comment]
review is up for 1.3: https://phabricator.mesosphere.com/D431

**Comment by meichstedt** [comment]
Fixed via ab0269f1841d8b46c85ad5038f3de82a69863b0f
